SonarQube.Common/AnalysisConfig/AnalysisConfig.cs

```diff
@@ -61,6 +61,11 @@ public class AnalysisConfig
         /// </summary>
         public AnalysisProperties LocalSettings { get; set; }
 
+        /// <summary>
+        /// Configuration for Roslyn analysers
+        /// </summary>
+        public AnalyzerSettings AnalyzerSettings { get; set; }
```

Comment:
We need the compiler settings during the build phase so I've moved the data class from the PreProcessor project to the Common project so it can be used by a new build task.
I renamed the data class from "AnalyzerCompilerConfig" to "AnalyzerSettings" to fit in with the naming of the other child classes of AnalysisConfig.